### ISSUE TITLE
[java] A false negative about PMD rule HardCodedCryptoKey

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3540]
**Affects PMD Version:6.39.0** 

**Rule:HardcodedCryptoKey**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.39.0/pmd_rules_java_security.html#hardcodedcryptokey>

**Description:**
PMD should have reported an critical warning to detect the bug at line 3 because when str is assigned by a literal string constant, the program will exist a security flaw about secrey key leakage. I think we can consider this situation specifically and it can be easily fixed.

**Code Sample demonstrating the issue:**

```java
void encrypt() {
    final String var0;
    var0 = "hard coded key here";
    SecretKeySpec keySpec =  new SecretKeySpec(var0.getBytes(), "AES");
}
```

**Expected outcome: A warning**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
